packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -92,9 +92,19 @@ export interface VirtualNodeListener {
   readonly portMapping?: PortMapping;
 
   /**
-   * Array fo HealthCheckProps for the node(s)
+   * Array of HealthCheckProps for the node(s)
    *
    * @default - no healthcheck
    */
   readonly healthCheck?: HealthCheck;
 }
+
+/**
+ * Configuration for Envoy Access logs for mesh endpoints
+ */
+export interface AccessLog {
+  /**
+   * Path to a file to write access logs to
+   */
+  readonly filePath: string;
```

Comment:
Question (for @skinny85 I think): We represent the options in this struct as a "union" in our APIs, which means only one value (e.g. `filePath` or something else) can be provided. In the future if we add a new option that's not a `filePath`, is this the right interface/approach?

Reply:
If that is the case, we have a different pattern in the CDK for that: the union-like class. Basically, you type this property as an abstract class:

```ts
export abstract class AccessLog {
  // ...
}

export interface VirtualNodeBaseProps {
  // .. 

  readonly accessLog?: AccessLog;
}
```

And then have static factory methods that create instances of it:

```ts
export abstract class AccessLog {
  public static fromFilePath(filePath: string): AccessLog {
    // ...
  }

  // ...
}
```

Does this make sense?